sklearn/linear_model/_ridge.py

```diff
@@ -1285,7 +1288,7 @@ def _eigen_decompose_gram(self, X, y, sqrt_sw):
             # containing the square roots of the sample weights.
             # by centering, it is orthogonal to the other columns
             K += np.outer(sqrt_sw, sqrt_sw)
-        eigvals, Q = linalg.eigh(K)
+        eigvals, Q = linalg.eigh(K, check_finite=False)
```

Comment:
Since `K` can be `+=np.outer(sqrt_sw, sqrt_sw)`, `K` may not be finite even if both `K` and `sqrt_sw` was:

```python
from sklearn.utils.extmath import safe_sparse_dot
import numpy as np

f_max = np.finfo(np.float64).max

K = np.full((2, 2), fill_value=f_max * 0.9)

K += f_max * 0.4
K
# array([[inf, inf],
#       [inf, inf]])
```